pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/ApexTreeBuilder.kt

```diff
@@ -424,8 +459,184 @@ class ApexTreeBuilder(val sourceCode: String, val parserOptions: ApexParserOptio
     private fun buildSizedArrayInitializer(node: SizedArrayInitializer) =
         ASTNewListInitExpression(node).apply { buildChildren(node, parent = this) }
 
+    /** Builds an [ApexNode] wrapper for the [DmlStatement]. */
+    private fun buildDmlStatement(node: DmlStatement) =
+        when (node) {
+            is DmlStatement.Insert -> ASTDmlInsertStatement(node)
+            is DmlStatement.Update -> ASTDmlUpdateStatement(node)
+            is DmlStatement.Delete -> ASTDmlDeleteStatement(node)
+            is DmlStatement.Undelete -> ASTDmlUndeleteStatement(node)
+            is DmlStatement.Upsert -> ASTDmlUpsertStatement(node)
+            is DmlStatement.Merge -> ASTDmlMergeStatement(node)
+        }.apply { buildChildren(node, parent = this) }
+
+    /** Wraps the body of a control statement with an [ASTBlockStatement] if it isn't already one. */
+    private fun wrapBody(body: Statement, parent: ApexNode<*>) =
+        when (body) {
+            is CompoundStatement -> build(body, parent) as ASTBlockStatement
+            else -> ASTBlockStatement(body).apply { buildAndSetParent(body, parent = this) }
+        }
+
+    /** Builds an [ASTIfElseBlockStatement] wrapper for the [IfStatement]. */
+    private fun buildIfStatement(node: IfStatement): ASTIfElseBlockStatement {
+        val (ifBlocks, elseBlock) = flattenIfStatement(node)
```

Comment:
I'd like to understand if statement flattening better. Perhaps it would be easier to discuss 'live'.